pkg/agent/plugin/nodeattestor/aws/iid_test.go

```diff
@@ -0,0 +1,192 @@
+package aws
+
+import (
+	"context"
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	jwt "github.com/dgrijalva/jwt-go"
+	"github.com/spiffe/spire/pkg/common/plugin/aws"
+	"github.com/spiffe/spire/proto/agent/nodeattestor"
+	"github.com/spiffe/spire/proto/common/plugin"
+	"github.com/stretchr/testify/suite"
+)
+
+func (s *Suite) buildDefaultIIDDocAndSig() (docBytes []byte, sigBytes []byte) {
+	// doc body
+	doc := aws.InstanceIdentityDocument{
+		AccountID:  "test-account",
+		InstanceID: "test-instance",
+		Region:     "test-region",
+	}
+	docBytes, err := json.Marshal(doc)
+	s.Require().NoError(err)
+
+	rng := rand.Reader
+	bitSize := 2048
+	key, err := rsa.GenerateKey(rng, bitSize)
+
+	// doc signature
+	docHash := sha256.Sum256(docBytes)
+	sig, err := rsa.SignPKCS1v15(rng, key, crypto.SHA256, docHash[:])
+	s.Require().NoError(err)
+
+	return docBytes, sig
+}
+
+func TestIIDAttestorPlugin(t *testing.T) {
+	suite.Run(t, new(Suite))
+}
+
+type Suite struct {
+	suite.Suite
+
+	p       *nodeattestor.BuiltIn
+	server  *httptest.Server
+	status  int
+	docBody string
+	sigBody string
+}
+
+func (s *Suite) SetupTest() {
+	s.server = httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		w.WriteHeader(s.status)
+
+		if strings.Contains(req.URL.String(), defaultIdentityDocumentPath) {
```

Comment:
might do a switch around `req.URL.Path` with two cases for each of the expected paths, and a default case that fails the call.